File: server/src/entities/Account.ts
import { Column, Entity, OneToMany, PrimaryGeneratedColumn } from 'typeorm';
import { Upload } from './Upload';

export enum AccountAmountsType {
  NEGATIVE_AMOUNT_EXPENSE = 'negamtexp',
  POSITIVE_AMOUNT_EXPENSE = 'posamtexp',
  SEPARATE_TYPE_COLUMN = 'septypecol',
}

@Entity()
export class Account {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({
    unique: true,
  })
  name: string;

  @Column()
  dateHeader: string;

  @Column()
  descriptionHeader: string;

  @Column()
  amountHeader: string;

  @Column({ nullable: true })
  typeHeader: string;

  @Column('numeric', { default: 0, precision: 12, scale: 2 })
  startingAmount: number;

  @Column('numeric', { default: 0, precision: 12, scale: 2 })
  balance: number;

  @Column({
    type: 'enum',
    enum: AccountAmountsType,
  })
  amountsType: AccountAmountsType;

  @Column({ nullable: true })
  color: string;

  @OneToMany(() => Upload, (upload) => upload.account)
  uploads: Upload[];
}

File: go-server/model/account.go
package model

import (
	"gorm.io/gorm"
)

const (
	// A negative value in the amount column indicates an expense
	NegativeAmountExpense = "negamtexp"

	// A positive value in the amount column indicates an expense
  PositiveAmountExpense = "posamtexp"

	// There is a separate column specifying whether the amount is income or expense
  SeparateTypeColumn = "septypecol"

	// There are separate columns for income vs expense
  SeparateIncomeExpenseColumns = "sepincexp"
)

type Account struct {
  gorm.Model
	Name string `gorm:"unique;not null;default:null"`
	DateHeader string `gorm:"not null;default:null"`
	DescriptionHeader string `gorm:"not null;default:null"`
	AmountHeader *string 
	IncomeHeader *string
	ExpenseHeader *string
	TypeHeader *string
	StartingAmount int64 `gorm:"not null;default:0"`
	Balance int64 `gorm:"not null;default:0"`
	AmountsType string `gorm:"not null;default null"` // TODO: can this be an enum
	Color *string
	Uploads []Upload
}